src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarHeader.Read.cs

```diff
@@ -0,0 +1,581 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Buffers;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.IO;
+using System.Text;
+
+namespace System.Formats.Tar
+{
+    // Reads the header attributes from a tar archive entry.
+    internal partial struct TarHeader
+    {
+        private const string UstarPrefixFormat = "{0}/{1}"; // "prefix/name"
+
+        // Attempts to read all the fields of the next header.
+        // Throws if end of stream is reached or if any data type conversion fails.
+        // Returns true if all the attributes were read successfully, false otherwise.
+        internal bool TryGetNextHeader(Stream archiveStream, bool copyData)
+        {
+            // The four supported formats have a header that fits in the default record size
+            byte[] rented = ArrayPool<byte>.Shared.Rent(minimumLength: TarHelpers.RecordSize);
```

Comment:
RecordSize is 512 bytes.  This can just be stackalloc'd rather than using ArrayPool, e.g.
```C#
Span<byte> buffer = stackalloc byte[TarHelpers.RecordSize];
buffer.Clear();
```